testing/test_rx_tx/usrp_setup.py

```diff
@@ -0,0 +1,201 @@
+		:param main_subdev: String representing the subdevice(s) for the main reciever
+		"""
+		rx_subdev = uhd.usrp.SubdevSpec(main_subdev)
+		self.usrp.set_rx_subdev_spec(rx_subdev)
+
+	def set_rx_rate(self, rx_rate):
+		"""
+		Sets the data rate for the reciever
+		:param rx_rate: The reciever data rate
+		"""
+		self.usrp.set_rx_rate(rx_rate)
+
+	def get_rx_rate(self, channel):
+		"""
+		Gets the reciever rate on a specified channel
+		:param channel: The desired channel
+		"""
+		return self.usrp.get_rx_rate(np.uint32(channel))
+
+	def set_rx_center_freq(self, freq, chans):
+		"""
+		Tunes the reciever to a desired frequency
+		:param freq: The desired reciever center frequency
+		:param chans: The channels to tune to freq
+		"""
+		rx_tune_request = uhd.types.TuneRequest(freq)
+		for channel in chans:
+			self.usrp.set_rx_freq(rx_tune_request, channel)
+			actual_freq = self.usrp.get_rx_freq(channel)
+			if not (actual_freq == freq):
+				print("Requested rx center frequency:", freq, "actual frequency:",
+						actual_freq, "\n")
+	def get_rx_center_freq(self, chan):
+		"""
+		Gets the center frequency for a specified channel
+		:param chan: The channel at which to retrieve the center frequency
+		"""
+		return self.usrp.get_rx_freq(chan)
+
+	def create_rx_stream(self, cpu, otw, chans):
+		"""
+		Sets up an rx streaming object based on given options
+		:par
+		:param cpu: The host cpu format as a string
+		:param otw: The over the wire format as a string
+		:param chans: Desired receiving channels
+		"""
+		rx_stream_args = uhd.usrp.StreamArgs(cpu, otw)
+		rx_stream_args.channels = chans
+		rx_stream = self.usrp.get_rx_stream(rx_stream_args)
+		return rx_stream
+
+	def setup_gpio(self, gpio_bank):
+		"""
+		Configures the given gpio bank for the txio
+		:param gpio_bank: String representing the gpio bank
+		"""
+		for i in np.arange(self.usrp.get_num_mboards()):
+			self.usrp.set_gpio_attr(gpio_bank, "CTRL", 0xFFFF, 0b11111111, i)
+			self.usrp.set_gpio_attr(gpio_bank, "DDR", 0xFFFF, 0b11111111, i)
+
+			# Mirror pins along bank for easier scoping
+			self.usrp.set_gpio_attr(gpio_bank, "ATR_RX", 0xFFFF, 0b000000010, i)
+			self.usrp.set_gpio_attr(gpio_bank, "ATR_RX", 0xFFFF, 0b000000100, i)
+
+			self.usrp.set_gpio_attr(gpio_bank, "ATR_TX", 0xFFFF, 0b000001000, i)
+			self.usrp.set_gpio_attr(gpio_bank, "ATR_TX", 0xFFFF, 0b000010000, i)
+
+			#XX is the actual TR signal
+			self.usrp.set_gpio_attr(gpio_bank, "ATR_XX", 0xFFFF, 0b000100000, i)
+			self.usrp.set_gpio_attr(gpio_bank, "ATR_XX", 0xFFFF, 0b001000000, i)
+
+			#0X acts as 'scope sync'
+			self.usrp.set_gpio_attr(gpio_bank, "ATR_0X", 0xFFFF, 0b010000000, i)
+			self.usrp.set_gpio_attr(gpio_bank, "ATR_0X", 0xFFFF, 0b100000000, i)
+
+	def setup(self):
+		"""
+		main method for handling board setup based on given config
+		file, frequencies, and channels
+		"""
+		# Configure USRP clock and gpio bank
+		self.set_usrp_clock_source(self.options.get_pps)
```

Comment:
Wrong arg to this fn. You need to pass in the 10MHz reference source. You are also missing a function to set the time source for pps.